webapp/STYLE_GUIDE.md

```diff
@@ -0,0 +1,157 @@
+- **Import**: Import the component's .scss file directly into the component's .tsx file (e.g., `import './header_footer_route.scss';`).
+- **Scope & Naming**:
+  - **Root**: Scope all styles under a unique root class named after the component (e.g., .header-footer-route).
+  - **BEM**: Use BEM-inspired naming in kebab-case. For root classes use PascalCase for blocks to match with the component name.
+  - **Elements**: Use the &__element suffix (e.g., .system-policy-indicator__title).
+  - **Modifiers**: Use modifier classes for variants (e.g., &.compact).
+- **Theme Tokens (CSS Variables)**:
+  - **Colors**: Always use CSS custom properties for colors (e.g., `color: var(--link-color);`). Do not hard-code hex values.
+  - **Transparency**: For transparent colors, use the RGB-tuple variable (e.g., `background: rgba(var(--sidebar-header-text-color-rgb), 0.08);)`.
+- **Layout**: Prefer display: flex for layouts; use `display: grid` for more complex grids
+- **Win specificity by:**
+  - Scoping under the component root class
+  - Using variant classes (e.g., .Component--compact)
+  - If you must override third-party/legacy, create a narrowly scoped wrapper class instead of !important.
+
+## Internationalization (I18n)
+
+- **Component vs. Hook**: Prefer the <FormattedMessage> component over the useIntl().formatMessage hook, unless you specifically need a string for a prop.
+- **Redux**: Actions or Redux state should store MessageDescriptor objects, with formatting deferred to the React component.
+- **Prohibited APIs**: Do not use localizeMessage or FormattedMarkdownMessage.
+- **Concatenation**: Use React Intl's Rich Text Formatting feature instead of concatenating translated strings.
+
+## Plugin Development
+
+- **APIs**: When exposing components or APIs for plugins, be extra sure it's necessary and limit the number of props to reduce the risk of breakage.
+- **Libraries**: Do not add new libraries for plugins to consume.
+
+## Testing
+
+- **Library**: Use React Testing Library (RTL) instead of Enzyme for all component unit tests.
+- **Helpers**: Import testing functions from utils/react_testing_utils and use renderWithContext for components that need Redux, I18n, or Router context.
+- **Method**: Test the functionality and behavior of the component. Do not use snapshot tests.
+- **Selectors**: Prefer accessible selectors in this order: `getByRole` > `getByLabelText` > `getByPlaceholderText` > `getByText` > `getByTestId` (last resort).
+- **Keyboard Testing**: Use `userEvent.keyboard('{Enter}')` and `userEvent.tab()` to test keyboard interactions.
+- **ARIA Verification**: Assert ARIA attributes are correct (e.g., `expect(el).toHaveAttribute('aria-expanded', 'true')`).
+
+## Asset Management & Constants
+
+- **Asset Location**: Store new images under webapp/channels/src/images.
+- **New Constants**: Add new preference keys, modal IDs, and enumerations to utils/constants.tsx.
```

Comment:
I'm going to remove these ones since these are actually things I want to get away from. I think it's better to put these closer to where they're used because it avoids us forgetting to remove assets that are no longer used, and those shared files in `utils` often are a nightmare for code splitting (although `constants.tsx` is nowhere near as bad as `utils.tsx`)